sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/tasks/TaskManagementSteps.java

```diff
@@ -399,6 +411,225 @@ public TaskManagementSteps(
                           .createdBy(tableRow.get(ColumnHeaders.CREATED_BY.toString()))
                           .build()));
         });
+
+    And(
+        "I click Export button in Task Directory",
+        () -> {
+          webDriverHelpers.waitForPageLoadingSpinnerToDisappear(40);
+          TimeUnit.SECONDS.sleep(2);
+          webDriverHelpers.clickOnWebElementBySelector(TASK_EXPORT_BUTTON);
+        });
+
+    When(
+        "I click on the Detailed Task Export button",
+        () -> {
+          String file = "./downloads/sormas_tasks_" + LocalDate.now().format(formatter) + "_.csv";
+
+          Path file_path = Paths.get(file);
+          if (webDriverHelpers.isFileExists(file_path)) {
+            Files.delete(file_path);
+          }
+          TimeUnit.SECONDS.sleep(8); // wait for basic download if in parallel
+          webDriverHelpers.clickOnWebElementBySelector(DETAILED_EXPORT_BUTTON);
+          TimeUnit.SECONDS.sleep(8); // wait for download start
+          webDriverHelpers.waitForFileExists(file_path, 90);
+        });
+
+    When(
+        "I click on the Custom Event Export button",
+        () -> {
+          TimeUnit.SECONDS.sleep(8); // wait for basic download if in parallel
```

Comment:
remove sleeps, just click on download button and wait after file to be downloaded (with the assert approach)